CMakeLists.txt

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2010-2020 The Regents of the University of Michigan
+# This file is from the freud project, released under the BSD 3-Clause License.
+
+cmake_minimum_required(VERSION 3.16)
+
+project(freud VERSION 2.3.0 LANGUAGES C CXX)
+
+set(DEFAULT_BUILD_TYPE "Release")
+
+# Set a default build type if none was specified
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to '${DEFAULT_BUILD_TYPE}' since none specified.")
+  set(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}" CACHE
+      STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+# Enable C++11
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_C_COMPILER $ENV{CC})
+set(CMAKE_CXX_COMPILER $ENV{CXX})
+
+# Enable warnings for gcc and clang
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+endif()
+
+# Enable color output from compiler
+if (CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=always")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fdiagnostics-color=always")
+endif()
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fcolor-diagnostics")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fcolor-diagnostics")
+endif()
+
+# Export compile_commands.json
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# External C++ dependencies
+find_package(TBB REQUIRED)
+#add_subdirectory(extern)
+include_directories("extern")
+#include_directories("${EIGEN_INCLUDE_DIR}"
+#                    "${DSET_INCLUDE_DIR}"
+#                    "${FSPH_INCLUDE_DIR}"
+#                    "${VOROPP_INCLUDE_DIR}")
+
+# Python / Cython dependencies
+#find_package(Python3 COMPONENTS NumPy REQUIRED)
+#message("${Python3_NumPy_INCLUDE_DIRS}")
+
+# C++ includes
+include_directories("${CMAKE_SOURCE_DIR}/cpp"
+                    "${CMAKE_SOURCE_DIR}/cpp/box"
+                    "${CMAKE_SOURCE_DIR}/cpp/cluster"
+                    "${CMAKE_SOURCE_DIR}/cpp/density"
+                    "${CMAKE_SOURCE_DIR}/cpp/diffraction"
+                    "${CMAKE_SOURCE_DIR}/cpp/environment"
+                    "${CMAKE_SOURCE_DIR}/cpp/locality"
+                    "${CMAKE_SOURCE_DIR}/cpp/order"
+                    "${CMAKE_SOURCE_DIR}/cpp/parallel"
+                    "${CMAKE_SOURCE_DIR}/cpp/pmft"
+                    "${CMAKE_SOURCE_DIR}/cpp/util")
+
+# Define library
+add_library(freud SHARED
```

Comment:
I believe voro++'s source files need to be in this list also.

Reply:
This is covered by `include_directories("extern")` and the corresponding `CMakeLists.txt` in that directory.